contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,448 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol" as Ownable;
+
+contract SeriesIndexer is Ownable.Ownable {
+    // Counter
+    uint256 private nextSeriesID = 1;
+    uint256 private nextArtistID = 1;
+
+    struct Series {
+        string metadata;
+        string contractTokenData;
+        uint256[] artistIDs;
+    }
+
+    // Artist Management
+    mapping(uint256 => address) private artistIDToAddress;
+    mapping(address => uint256) private addressToArtistID;
+    mapping(uint256 => bool) private ownerRightsRevokedForArtistID;
```

Comment:
I think this should be on series level instead of artist level. We could also provide a feature for revoking all series from an artist.